weatherApp/ViewControllers/LocationsViewController.swift

```diff
@@ -0,0 +1,62 @@
+import UIKit
+
+protocol PassCityInfo {
+    func passCoordinate(latitude: Double, longitude: Double)
+}
+
+class LocationsViewController: UIViewController {
+    
+    @IBOutlet weak var searchCity: UISearchBar!
+    @IBOutlet weak var locationTableView: UITableView!
+    
+    var locationList = [dhaka, rajshahi, khulna, rangpur, barishal, chattogram, sylhet]
+    var searchData = [city]()
+    var delegate: PassCityInfo!
```

Comment:
Don't use force unwrap. Make it optional.
Always make the delegate as weak property otherwise you might get stuck in retain cycle.

```suggestion
    weak var delegate: PassCityInfo?
```